akka-stream/src/main/scala/akka/stream/BoundedQueueSource.scala

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.stream
+
+/**
+ * A queue of the given size that gives immediate feedback whether an element could be enqueued or not.
+ */
+trait BoundedQueueSource[T] {
```

Comment:
Why not `BoundedSourceQueue`? After all, it's not a source but a queue and that would also be consistent with the existing naming? (I just realized that I confused those also above in my docs suggestions).

Reply:
Btw. I think it would be fine if a `BoundedQueueSource(Stage)` would provide a `BoundedSourceQueue`. After all these are two opposite ends of looking at the component.

Reply:
I renamed `BoundedQueueSource` to `BoundedSourceQueue`.  I changed the stage too just to make it consistent.